source/applications/advanced/color_balance.py

```diff
@@ -64,26 +83,60 @@ def _compute_mean_rgb(rgb, pixels):
     pixel_hight_end_index = int((height + pixels) / 2)
     pixel_width_start_index = int((width - pixels) / 2)
     pixel_width_end_index = int((width + pixels) / 2)
-    red = rgb[
+    return rgb[
         pixel_hight_start_index:pixel_hight_end_index,
         pixel_width_start_index:pixel_width_end_index,
-        0,
+        :,
     ]
-    green = rgb[
-        pixel_hight_start_index:pixel_hight_end_index,
-        pixel_width_start_index:pixel_width_end_index,
-        1,
-    ]
-    blue = rgb[
-        pixel_hight_start_index:pixel_hight_end_index,
-        pixel_width_start_index:pixel_width_end_index,
-        2,
-    ]
-    mean_red = np.mean(np.reshape(red, -1), dtype=np.float64)
-    mean_green = np.mean(np.reshape(green, -1), dtype=np.float64)
-    mean_blue = np.mean(np.reshape(blue, -1), dtype=np.float64)
 
-    return MeanColor(red=mean_red, green=mean_green, blue=mean_blue)
+
+def _compute_mean_rgb(rgb):
+    """Compute mean RGB values.
+
+    Args:
+        rgb: RGB image
+
+    Returns:
+        mean_color: RGB channel mean values
+
+    """
+    mean_rgb = np.mean(np.mean(rgb, axis=0), axis=0)
+    return MeanColor(red=mean_rgb[0], green=mean_rgb[1], blue=mean_rgb[2])
+
+
+def _get_tuned_settings(camera: zivid.Camera) -> zivid.Settings2D:
+    """Get settings such that max mean(RGB) is < 240. In other words, no saturation on average.
+
+    Args:
+        camera: Zivid camera
+
+    Returns:
+        settings_2d: 2D capture settings without saturation on average
+
+    """
+    settings_2d = zivid.Settings2D(
+        acquisitions=[
+            zivid.Settings2D.Acquisition(
+                aperture=5.66,
+                exposure_time=datetime.timedelta(microseconds=80000),
+                brightness=0.0,
```

Comment:
Should we be explicit here on not using the projector?

Reply:
What do you mean? Add comments?

Reply:
No, I meant that we maybe should not hard code it to 0 but something else. It could be that someone is using the camera in e.g. strong yellow light with the projector. Cases like that may be more common than those where the projector is off.

Reply:
Maybe highlight these settings in main(), making a point that these should be set manually? I also thought about whether to update the automatic setting function to limit the Aperture and use Exposure Time instead and maybe even Gain.